test_utils/src/mock_server.rs

```diff
@@ -35,54 +36,106 @@ pub struct MockServerState {
     pub host: String,
 }
 
-pub type SharedMockServerState = Arc<MockServerState>;
+#[derive(Debug, Clone)]
+pub struct MockServerBuilder {
+    repo_create_handler: MethodRouter<SharedMockServerState>,
+    create_bundle_handler: MethodRouter<SharedMockServerState>,
+    get_quarantining_config_handler: MethodRouter<SharedMockServerState>,
+    s3_upload_handler: MethodRouter<SharedMockServerState>,
+}
+
+impl MockServerBuilder {
+    pub fn new() -> Self {
+        Self {
+            repo_create_handler: post(repo_create_handler),
+            create_bundle_handler: post(create_bundle_handler),
+            get_quarantining_config_handler: post(get_quarantining_config_handler),
+            s3_upload_handler: put(s3_upload_handler),
+        }
+    }
```

Comment:
allows setting the route handlers for doing things like failed API calls, slow API calls, etc.